sysboundary/sysboundary.cpp

```diff
@@ -646,62 +492,66 @@ bool SysBoundary::classifyCells(dccrg::Dccrg<spatial_cell::SpatialCell, dccrg::C
    }
 
    technicalGrid.updateGhostCells();
-
-   return success;
 }
 
-/*!\brief Apply the initial state to all system boundary cells.
- * Loops through all SysBoundaryConditions and calls the corresponding applyInitialState
- * function. This function must apply the initial state for all existing particle species.
- *
+/*!\brief Apply the initial state to all boundary cells.
+ * Loops through all BoundaryConditions and apply the initial state for all
+ * existing particle species.
  * \param mpiGrid Grid
- * \retval success If true, the application of all system boundary states succeeded.
  */
-bool SysBoundary::applyInitialState(dccrg::Dccrg<SpatialCell, dccrg::Cartesian_Geometry>& mpiGrid,
-                                    FsGrid<array<Real, fsgrids::bfield::N_BFIELD>, FS_STENCIL_WIDTH>& perBGrid,
+void SysBoundary::applyInitialState(dccrg::Dccrg<SpatialCell, dccrg::Cartesian_Geometry>& mpiGrid,
+                                    FsGrid<std::array<Real, fsgrids::bfield::N_BFIELD>, FS_STENCIL_WIDTH>& perBGrid,
                                     Project& project) {
-   bool success = true;
-
    list<SBC::SysBoundaryCondition*>::iterator it;
    for (it = sysBoundaries.begin(); it != sysBoundaries.end(); it++) {
-      if (                                // This is to skip the reapplication
-          Parameters::isRestart           // When not restarting
-          && !(*it)->doApplyUponRestart() // When reapplicaiton is not requested
-      ) {
+      // Skip when not restarting or not requested.
+      if (Parameters::isRestart && !(*it)->doApplyUponRestart()) {
          continue;
       }
-      if (!(*it)->applyInitialState(mpiGrid, perBGrid, project)) {
-         cerr << "ERROR: " << (*it)->getName() << " system boundary condition initial state not applied correctly." << endl;
-         success = false;
-      }
+      (*it)->applyInitialState(mpiGrid, perBGrid, project);
    }
+}
 
-   return success;
+void SysBoundary::updateState(const dccrg::Dccrg<SpatialCell, dccrg::Cartesian_Geometry>& mpiGrid,
+                              FsGrid<std::array<Real, fsgrids::bfield::N_BFIELD>, FS_STENCIL_WIDTH>& perBGrid,
+                              creal t) {
+   list<SBC::SysBoundaryCondition*>::iterator it;
+   if (anyDynamic) {
+      for (it = sysBoundaries.begin(); it != sysBoundaries.end(); it++) {
+         // Skip when not restarting or not requested.
+         if (Parameters::isRestart && !(*it)->doApplyUponRestart())
```

Comment:
IMPORTANT! 
Are you sure about this?!?
If called within a restarted run but not requesting reapplyuponrestart, then it is in effect turning dynamic boundary conditions off. 

In my opinion, reapplyuponrestart is only for non-dynamic boundary conditions, as... well, dynamic conditions should be updated regardless of it being a restart or not, so such a flag doesn't make sense for those!

Parameters::isRestart is not turned off after initial handling.

Also, at least in the past, reapplyuponrestart affected the call of applyInitialState().